src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -103,9 +128,44 @@ public void launch() {
             kernel.getContext().get(c).startup();
         }
 
+        final Queue<DeviceIdentityInterface> provisioningPlugins = findProvisioningPlugins();
         // Must be called before everything else so that these are available to be
         // referenced by main/dependencies of main
         final Queue<String> autostart = findBuiltInServicesAndPlugins(); //NOPMD
+        // run the provisioning if device is not provisioned
+        if (!kernel.getContext().get(DeviceConfiguration.class).isDeviceConfiguredToTalkToCloud()
+                && !provisioningPlugins.isEmpty()) {
+            RetryUtils.RetryConfig retryConfig = RetryUtils.RetryConfig.builder()
+                    .maxAttempt(MAX_PROVISIONING_PLUGIN_RETRY_ATTEMPTS)
+                    .retryableExceptions(Collections.singletonList(RetryableProvisioningException.class))
+                    .build();
+            ExecutorService executorService = kernel.getContext().get(ExecutorService.class);
+            executorService.submit(() -> {
```

Comment:
use `execute` if you're going to ignore the returned future. That will resolve the spotbugs error.